packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.types.ts

```diff
@@ -27,10 +27,11 @@ export interface DesktopDatePickerProps<
 > extends BaseDatePickerProps<TDate>,
     DesktopOnlyPickerProps {
   /**
-   * Years rendered per row.
-   * @default 4
+   * If `false`, display the years in their default order (chronological)
+   * If `true`, reverse the order of displayed years
+   * @default false
    */
-  yearsPerRow?: 3 | 4;
+  reverseYears?: boolean;
```

Comment:
Could we please remove this explicit prop from here and the `DesktopDateTimePicker.types` as well? 🤔

Reply:
Hi again @LukasTy. There were a few type files under pickers where yearsPerRow was specified directly on the main definition rather than incorporating it from ExportedYearCalendarProps that was in use on the YearCalendar.

Perhaps dependencies were not wanted since it's defined on another component, but it seems to me that the ExportedYearPicker definition could be used in those places. So I removed the prop I added and also refactored the existing setup slightly because the type definition was needed on DatePicker or the prop wasn't recognized as it was on YearCalendar.

If you can check the new type extensions I made to the pickers, those seem to be quite complicated from the start. At lease the demos seem to work correctly still, let's see if all the tests are passing still.